components/indexer.py

```diff
@@ -1,30 +1,35 @@
 from magicbot import feedback
+import wpilib
 
 
 class Indexer:
     indexer_motors: list
     indexer_switches: list
+    ready_piston: wpilib.DigitalInput
 
     def on_enable(self) -> None:
+        for motor in self.indexer_motors:
+            motor.setInverted(True)
         self.indexing = True
+        self.speed = 0.2
 
     def execute(self) -> None:
         if self.indexing:
-            motor_states = [True] * len(self.indexer_motors)
-            switch_results = [switch.get() for switch in self.indexer_switches]
-            # Test the back switch (because all others require 2 switches)
-            if not switch_results[0]:
-                motor_states[0] = False
-
-            # Disable motor if switch and previous switch are pressed
-            for i in range(1, len(motor_states)):
-                if not switch_results[i] and not switch_results[i - 1]:
-                    motor_states[i] = False
-
-            # Set all motors to required states
-            for motor_state, motor in zip(motor_states, self.indexer_motors):
-                if motor_state:
-                    motor.set(1)
+            for i, (motor, switch) in enumerate(
+                zip(
+                    self.indexer_motors,
+                    [switch.get() for switch in self.indexer_switches],
+                )
+            ):
+
+                if switch:
+                    if not i:
+                        if not self.ready_piston.get():
+                            motor.set(self.speed * 2)
+                        else:
+                            motor.stopMotor()
```

Comment:
This whole thing took me a solid 10 minutes to figure out what it was doing tonight. Why is there a special case specifically for the first motor, which happens to be the motor for the injector? Why does it get its own special speed if it's in this list? Why is there a list comprehension in the for loop's iterable?

Reply:
You really should put a comment in explaining why the first motor is skipped.
Consider rewriting the loop like this:
These lines go in setup, as they are run only once:
```
self.MAX_MOTOR_INDEX = len(self.indexer_motors)
self.faster = self.speed * 2 # Both this and the assignment to self.speed could go in init, to really run only once
```
Now your loop can look like this
```
# loop from 1 to skip the first motor
for i in [1,self.MAX_MOTOR_INDEX]:
  if self.indexer.switches[i].get():
     if not self.ready_piston.get()
        self.indexer_motors[i].set(faster)
     else:
       self.indexer_motors[i].stopMotor()
```
This is shorter and clearer. Skipping the first item happens in the loop control.